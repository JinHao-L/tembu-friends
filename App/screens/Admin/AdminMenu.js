import React, { Component } from 'react';
import { View, StyleSheet, ScrollView } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';

import { MenuButton } from '../../components';
import { Colors } from '../../constants';

class AdminMenu extends Component {
    navigating = false;
    goToReports = () => {
        if (this.navigating) {
            return;
        }
        this.navigating = true;
        setTimeout(() => (this.navigating = false), 500);
        return this.props.navigation.navigate('Reports');
    };

    goToUsers = () => {
        if (this.navigating) {
            return;
        }
        this.navigating = true;
        setTimeout(() => (this.navigating = false), 500);
        return this.props.navigation.navigate('Users');
    };

    render() {
        return (
            <View style={{ flex: 1 }}>
                <LinearGradient
                    colors={[Colors.appGreen, Colors.appLightGreen]}
                    style={styles.container}
                >
                    <ScrollView style={styles.contentContainer}>
                        <MenuButton
                            style={styles.adminButton}
                            textStyle={{ color: Colors.appGreen }}
                            onPress={this.goToUsers}
                        >
                            Users
                        </MenuButton>
                        <MenuButton
                            style={styles.adminButton}
                            textStyle={{ color: Colors.appGreen }}
                            onPress={this.goToReports}
                        >
                            Reported Posts
                        </MenuButton>
                    </ScrollView>
                </LinearGradient>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    contentContainer: {
        paddingTop: 15,
        marginHorizontal: 15,
    },
    adminButton: {
        height: 40,
        alignItems: 'center',
        backgroundColor: Colors.appWhite,
    },
});

export default AdminMenu;
