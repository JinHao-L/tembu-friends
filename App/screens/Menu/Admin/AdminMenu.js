import React, { Component } from 'react';
import { View, StyleSheet, SafeAreaView } from 'react-native';

import { MainText, MenuButton } from '../../../components';
import { Colors } from '../../../constants';
import { LinearGradient } from 'expo-linear-gradient';

class AdminMenu extends Component {
    state = {};

    goToReports = () => {
        return this.props.navigation.navigate('Reports');
    };

    goToUsers = () => {
        return this.props.navigation.navigate('Users');
    };

    render() {
        return (
            <SafeAreaView style={{ flex: 1 }}>
                <LinearGradient
                    colors={[Colors.appGreen, Colors.appLightGreen]}
                    style={styles.container}
                >
                    <View style={styles.header}>
                        <MainText style={styles.title}>Admin Privileges</MainText>
                    </View>
                    <View style={styles.contentContainer}>
                        <MenuButton
                            style={styles.adminButton}
                            textStyle={{ color: 'white' }}
                            onPress={this.goToUsers}
                        >
                            Users
                        </MenuButton>
                        <MenuButton
                            style={styles.adminButton}
                            textStyle={{ color: 'white' }}
                            onPress={this.goToReports}
                        >
                            Reports
                        </MenuButton>
                    </View>
                </LinearGradient>
            </SafeAreaView>
        );
    }
}

styles = StyleSheet.create({
    header: {
        paddingBottom: 10,
        paddingTop: 20,
    },
    container: {
        flex: 1,
    },
    contentContainer: {
        paddingTop: 15,
        marginHorizontal: 30,
    },
    title: {
        textAlign: 'left',
        color: 'white',
        fontSize: 24,
        left: 30,
    },
    adminButton: {
        alignItems: 'center',
        backgroundColor: Colors.appGreen,
    },
});

export default AdminMenu;
