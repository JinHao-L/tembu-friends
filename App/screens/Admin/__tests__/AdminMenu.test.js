import React from 'react';
import renderer from 'react-test-renderer';

import AdminMenu from '../AdminMenu';

describe('<AdminMenu/>', () => {
    const tree = renderer.create(<AdminMenu />).toJSON();
    it('has 1 child', () => {
        expect(tree.children.length).toBe(1);
    });

    it('renders correctly', () => {
        expect(tree).toMatchSnapshot();
    });
    //
    // it('has 2 menu items', () => {
    //     expect(tree.children.length).toBe(1);
    // });
});
