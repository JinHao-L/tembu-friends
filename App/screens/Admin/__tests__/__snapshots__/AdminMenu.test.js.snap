// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminMenu/> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4280583256,
          4281647232,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "marginHorizontal": 15,
          "paddingTop": 15,
        }
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "borderColor": "#D9D9D9",
                "borderRadius": 11,
                "borderWidth": 1,
                "margin": 1,
                "marginBottom": 15,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 4,
              },
              undefined,
            ]
          }
        >
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            onPress={[Function]}
            rippleColor={4292467161}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 10,
                    "flex": 1,
                    "justifyContent": "center",
                    "paddingHorizontal": 5,
                    "paddingVertical": 5,
                  },
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "height": 40,
                  },
                ],
              ]
            }
            underlayColor="#D9D9D9"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Montserrat-SemiBold",
                        "fontSize": 13,
                      },
                      Array [
                        Object {
                          "color": "#000000",
                          "fontSize": 18,
                          "textAlign": "left",
                          "textAlignVertical": "center",
                        },
                        Object {
                          "color": "#248458",
                        },
                      ],
                    ]
                  }
                >
                  Users
                </Text>
              </View>
            </View>
          </RNGestureHandlerButton>
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "#D9D9D9",
                "borderRadius": 11,
                "borderWidth": 1,
                "margin": 1,
                "marginBottom": 15,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 4,
              },
              undefined,
            ]
          }
        >
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            onPress={[Function]}
            rippleColor={4292467161}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 10,
                    "flex": 1,
                    "justifyContent": "center",
                    "paddingHorizontal": 5,
                    "paddingVertical": 5,
                  },
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "height": 40,
                  },
                ],
              ]
            }
            underlayColor="#D9D9D9"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Montserrat-SemiBold",
                        "fontSize": 13,
                      },
                      Array [
                        Object {
                          "color": "#000000",
                          "fontSize": 18,
                          "textAlign": "left",
                          "textAlignVertical": "center",
                        },
                        Object {
                          "color": "#248458",
                        },
                      ],
                    ]
                  }
                >
                  Reported Posts
                </Text>
              </View>
            </View>
          </RNGestureHandlerButton>
        </View>
      </View>
    </RCTScrollView>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
